api/apis/route_handler.go

```diff
@@ -290,9 +229,143 @@ func (h *RouteHandler) routeCreateHandler(w http.ResponseWriter, r *http.Request
 	router.Path(RouteGetEndpoint).Methods("GET").HandlerFunc(h.routeGetHandler)
 	router.Path(RouteGetListEndpoint).Methods("GET").HandlerFunc(h.routeGetListHandler)
 	router.Path(RouteGetDestinationsEndpoint).Methods("GET").HandlerFunc(h.routeGetDestinationsHandler)
 	router.Path(RouteCreateEndpoint).Methods("POST").HandlerFunc(h.routeCreateHandler)
+	router.Path(RouteAddDestinationsEndpoint).Methods("POST").HandlerFunc(h.routeAddDestinationsHandler)
+}
+
+// Fetch Route and compose related Domain information within
+func (h *RouteHandler) lookupRouteAndDomain(ctx context.Context, routeGUID string) (repositories.RouteRecord, error) {
+	// TODO: Instantiate config based on bearer token
+	// Spike code from EMEA folks around this: https://github.com/cloudfoundry/cf-crd-explorations/blob/136417fbff507eb13c92cd67e6fed6b061071941/cfshim/handlers/app_handler.go#L78
+	client, err := h.buildClient(h.k8sConfig)
+	if err != nil {
+		return repositories.RouteRecord{}, err
+	}
+
+	route, err := h.routeRepo.FetchRoute(ctx, client, routeGUID)
+	if err != nil {
+		return repositories.RouteRecord{}, err
+	}
+
+	domain, err := h.domainRepo.FetchDomain(ctx, client, route.Domain.GUID)
+	// We assume K8s controller will ensure valid data, so the only error case is due to eventually consistency.
+	// Return a generic retryable error.
+	if err != nil {
+		err = errors.New("resource not found for route's specified domain ref")
+		return repositories.RouteRecord{}, err
+	}
+
+	route = route.UpdateDomainRef(domain)
+
+	return route, nil
+}
+
+func (h *RouteHandler) lookupRouteAndDomainList(ctx context.Context) ([]repositories.RouteRecord, error) {
+	// TODO: Instantiate config based on bearer token
+	// Spike code from EMEA folks around this: https://github.com/cloudfoundry/cf-crd-explorations/blob/136417fbff507eb13c92cd67e6fed6b061071941/cfshim/handlers/app_handler.go#L78
+	client, err := h.buildClient(h.k8sConfig)
+	if err != nil {
+		return []repositories.RouteRecord{}, err
+	}
+
+	routeRecords, err := h.routeRepo.FetchRouteList(ctx, client)
+	if err != nil {
+		return []repositories.RouteRecord{}, err
+	}
+
+	domainGUIDToDomainRecord := make(map[string]repositories.DomainRecord)
+
+	for i, routeRecord := range routeRecords {
+		currentDomainGUID := routeRecord.Domain.GUID
+		domainRecord, has := domainGUIDToDomainRecord[currentDomainGUID]
+		if !has {
+			domainRecord, err = h.domainRepo.FetchDomain(ctx, client, currentDomainGUID)
+			if err != nil {
+				err = errors.New("resource not found for route's specified domain ref")
+				return []repositories.RouteRecord{}, err
+			}
+			domainGUIDToDomainRecord[currentDomainGUID] = domainRecord
+		}
+		routeRecords[i] = routeRecord.UpdateDomainRef(domainRecord)
+	}
+
+	return routeRecords, nil
+}
+
+func getDomainsForRoutes(ctx context.Context, domainRepo CFDomainRepository, client client.Client, routeRecords []repositories.RouteRecord) ([]repositories.RouteRecord, error) {
```

Comment:
Is this function still used anywhere?

Reply:
Yes, it is used here https://github.com/cloudfoundry/cf-k8s-controllers/blob/5b0dc1a2a9cbf2be31247eb03264de3f9be60eac/api/apis/app_handler.go#L572

It might be a bit confusing as this function is not related to the PR but shows up as a change because we did a bit of formatting and moved all helper functions in this file to the very bottom.